collect_app/src/main/java/org/odk/collect/android/tasks/DownloadFormListTask.java

```diff
@@ -401,7 +401,7 @@ private boolean isThisFormAlreadyDownloaded(String formId) {
     }
 
     private boolean isNewerFormVersionAvailable(String md5Hash) {
-        return new FormsDao().getFormsCursorForMd5Hash(md5Hash).getCount() == 0;
+        return md5Hash != null && new FormsDao().getFormsCursorForMd5Hash(md5Hash).getCount() == 0;
```

Comment:
I think the version check should take the formID into account. Consider this scenario:
- I have a form with formID form1 and file contents with hash1
- I have a second form with formID form2 and file contents with hash2
- I change the form with formID form1 to have contents with hash2
form1 will not say it has an update available.

This may not be a super likely scenario but this check still seems very strange to me. It's not that we want to see if the hash is in the table at all. We want to see if a specific form's hash matches the remote one.

Reply:
But if you change the form with formID form1 to have contents with hash2 then both forms have the same id an version. It's this case https://github.com/opendatakit/collect/issues/1693 and we can't use both.

Reply:
The case I'm talking about is two different IDs but the same form contents. Since we're currently considering formID the differentiator for forms, the test here should ask whether the form with a matching formID has a matching hash rather than asking if the hash is in the table at all.

Reply:
But you can't have two forms with different id but the same hash. If you change the id then the hash is different too.

Reply:
Ah, because the form does have to contain a matching ID, doesn't it. Ok, I suppose that's ok, then.